apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/builtin/JvmGcMetrics.java

```diff
@@ -0,0 +1,92 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 Elastic and contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.metrics.builtin;
+
+import co.elastic.apm.agent.context.LifecycleListener;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.metrics.DoubleSupplier;
+import co.elastic.apm.agent.metrics.MetricRegistry;
+import com.sun.management.ThreadMXBean;
+import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class JvmGcMetrics implements LifecycleListener {
+
+    private final List<GarbageCollectorMXBean> garbageCollectorMXBeans = ManagementFactory.getGarbageCollectorMXBeans();
+
+    @Override
+    public void start(ElasticApmTracer tracer) {
+        bindTo(tracer.getMetricRegistry());
+    }
+
+    void bindTo(final MetricRegistry registry) {
+        for (final GarbageCollectorMXBean garbageCollectorMXBean : garbageCollectorMXBeans) {
+            final Map<String, String> tags = Collections.singletonMap("name", garbageCollectorMXBean.getName());
+            registry.addUnlessNegative("jvm.gc.count", tags, new DoubleSupplier() {
+                @Override
+                public double get() {
+                    return garbageCollectorMXBean.getCollectionCount();
```

Comment:
What would be more interesting - collection count or collection count since last measurement? For graphs, an ever increasing one is not informative as a mostly-stagnant one where glitches are easier to see. This is something that can be calculated in the server or Kibana as well, but since it is not a global metric, I don't think it would be.

Reply:
It's quite common to report counters and use a derivative aggregation to graph the deltas.